### ISSUE TITLE
Error handling

### ISSUE THREAD

**Issue by jmrox2001** [jmrox2001/Pay-calculator-experimental#2]
Line 70-92: I think this needs a better method of error handling, but I don't know how to do it better.

**Comment by mosheur15** [comment]
What is the error you're trying to catch?

**Comment by jmrox2001** [comment]
I am trying to catch errors with entering filing status. I believe it works correctly, but I don't understand this chunk of code enough. It was a copy and paste from my "tax rate tool" repository, so I don't know if this works correctly in this context.

**Comment by mosheur15** [comment]
as you mentioned line 70-92 looks fine. Still I did not understand how your algorithm should behave, so I did not catch any errors by reading it. 

But I would suggest you to comment how the process works. Then implement the code.
For example,
Write 5-10 lines of comments at the top, how you'll calculate the tax in a short summary.

Then add comments above each functions saying which part of your entire process they are solving. And how they are doing that.

For a bonus, name the variables as such they can have a meaning when you look through your code.
For example instead of naming a 'black cat' to 'bc' use camel case ('blackCat') or underscore ('black_cat').
It's more readable and you'll have better understanding of your code when you'll look at it a week later again.

By doing that not only your code will look more professional and understandable, you'll also get help from others easily.

**Comment by jmrox2001** [comment]
Yeah it’s confusing even for me. I will work on implementing your
suggestions and have an “update” maybe tonight or tomorrow. I’ll also try
to break down the algorithm more. My main goal was just making it work so
it is a bit thrown together.
